### DISCUSSION TITLE
Managing Buffers

### DISCUSSION THREAD

**Discussion by StevenXL** [LazyVim/LazyVim#2121] [Q&A] [answered]
I understand that LazyVim is opinionated when it comes to management of buffers. I appreciate that strong opinion!

However, it doesn't vibe with the way that I like to work. I was wondering how I could configure LazyVim so that it has the following behaviors:

1. Opening a file does so in the current window.
2. Closing a window is **not** equivalent to closing a buffer
3. I can list buffers in the order in which they were opened

Thanks folks!

**Comment by dpetka2001** [comment]
1. I'm not really sure what you mean here, but every file opens in its own buffer and `bufferline` shows the buffers at the top next to each other. Not sure what you're trying to achieve here.
2. Closing a window does not close the buffer. Buffers are shared between windows and tabs. Even if you close a tab, the buffer still won't be closed, until you explicitly do `<leader>bd` which is the equivalent to `:bdelete`.
3. You would have to change the keymap that LazyVim provides by default, which is `<leader>,` and `<leader>fb`, in your own configuration to whatever you want, because LazyVim uses `sort_mru=true sort_lastused=true` when showing the buffers listed.

**Comment by StevenXL** [comment]
Thanks @dpetka2001. I think the behavior I am seeing is that when I open a file (either through telescope or neotree), it opens in a new `tab`. I would rather it open in the current window. Am I mistaken here?

**Reply by dpetka2001** [reply, marked as answer]
What you're seeing is opening the file you choose in a new buffer. `bufferline` shows all the buffers that are open in Neovim at the top, which might seem like `tabs` in other editors, but they're actually buffers. You can create a new tab by pressing `<leader><tab><tab>` and you will see at the top right of your screen 2 numbers, 1 and 2. Those are the Neovim actual tabs. This is normal behavior. You could disable `bufferline` and then you would only see the active buffer on your screen. When you're opening a file, Neovim opens a new buffer for the file to be loaded in memory. It's just that `bufferline` shows you all the open buffers at the top left. It's just a visual aesthetic. Even if you disable `bufferline` files will still open each one in its own buffer, it's just that you will only be seeing the currently active buffer instead of having `bufferline` showing all of them. It's just a matter of preference. You can do whatever you want. Either disable `bufferline` to not have this visual aesthetic or keep it enabled and see all the available buffers shown.

Edit: read the official Neovim [docs](https://neovim.io/doc/user/windows.html) for more clarification.

**Reply by StevenXL** [reply]
Thank you! Disabling `bufferlline` was all I needed. My brain mapped that UI to tabs, and I couldn't get that out of my head until the UI was gone. Thanks again!
